python/lsst/analysis/drp/calcFunctors.py

```diff
@@ -143,3 +143,180 @@ def columns(self):
+    @property
+    def columns(self):
+        return (self.color1_flux1,
+                self.color1_flux2,
+                self.color2_flux1,
+                self.color2_flux2)
+
+    def __call__(self, df):
+        color1_flux1 = df[self.color1_flux1].values*u.Unit(self.color1_flux1_units)
+        color1_mag1 = color1_flux1.to(u.ABmag).value
+
+        color1_flux2 = df[self.color1_flux2].values*u.Unit(self.color1_flux2_units)
+        color1_mag2 = color1_flux2.to(u.ABmag).value
+
+        color2_flux1 = df[self.color2_flux1].values*u.Unit(self.color2_flux1_units)
+        color2_mag1 = color2_flux1.to(u.ABmag).value
+
+        color2_flux2 = df[self.color2_flux2].values*u.Unit(self.color2_flux2_units)
+        color2_mag2 = color2_flux2.to(u.ABmag).value
+
+        color1 = color1_mag1 - color1_mag2
+        color2 = color2_mag1 - color2_mag2
+
+        color_diff = color1 - color2
+
+        if self.return_millimags:
+            color_diff *= 1000.0
+
+        return color_diff
+
+
+class ColorDiffPull(ColorDiff):
+    """Calculate the difference between two colors, scaled by the color error;
+    Each color is derived from two flux columns.
+
+    The color difference is computed as (color1 - color2) with:
+
+    color1 = color1_mag1 - color1_mag2
+    color2 = color2_mag1 - color2_mag2
+
+    where color1_mag1 is the magnitude associated with color1_flux1, etc.
+
+    The color difference (color1 - color2) is then scaled by the error on
+    the color as computed from color1_flux1_err, color1_flux2_err,
+    color2_flux1_err, color2_flux2_err.  The errors on color2 may be omitted
+    if the comparison is between an "observed" catalog and a "truth" catalog.
+
+    Parameters
+    ----------
+    df : `pandas.core.frame.DataFrame`
+        The catalog to calculate the color difference from.
+
+    Returns
+    -------
+    The color difference scaled by the error.
+
+    Notes
+    -----
+    The flux columns need to be in units that can be converted
+    to janskies. This action doesn't have any calibration
+    information and assumes that the fluxes are already
+    calibrated.
+    """
+    color1_flux1_err = Field(doc="Error column for flux1 for color1",
+                             dtype=str)
+    color1_flux2_err = Field(doc="Error column for flux2 for color1",
+                             dtype=str)
+    color2_flux1_err = Field(doc="Error column for flux1 for color2",
+                             dtype=str,
+                             default="")
+    color2_flux2_err = Field(doc="Error column for flux2 for color2",
+                             dtype=str,
+                             default="")
+
+    @property
+    def columns(self):
+        cols = (self.color1_flux1,
+                self.color1_flux2,
+                self.color2_flux1,
+                self.color2_flux2,
+                self.color1_flux1_err,
+                self.color1_flux2_err)
+
+        if self.color2_flux1_err and self.color2_flux2_err:
```

Comment:
I don't quite follow why the errors for `color2` have to be treated separately this way.

Reply:
In this case, we're subtracting an observed color (with errors) from a true color (without errors).  I can foresee a future where we want to subtract an observed color from a separate observed color (both with errors).  But if you try to subtract (say) a true color from a true color, both without errors, then you divide by 0 and it won't work.  
If you prefer I could treat the color1 errors like this as well, and then just raise in the config validation if you configure such that neither color1 nor color2 has errors.

Reply:
I think that'd be good. Alternatively, a comment along this line would also be sufficient.